lib/pdn/pdn.c

```diff
@@ -117,6 +118,37 @@ static void on_cnec_esm(const char *notif)
 	}
 }
 
+static void on_cnec_emm(const char *notif)
+{
+	char *p;
+	uint32_t cid = 0;
+	uint32_t emm_err = UINT32_MAX;
+	struct pdn *pdn;
+	int ret;
+
+	/* +CNEC_EMM: <cause>,<cid> */
+	ret = sscanf(notif, "+CNEC_EMM: %d,%d", emm_err, cid);
+
+	/* CID is optional for EMM errors */
+	if (ret < 1) {
+		return;
+	}
+
+	pdn = pdn_find(cid);
+	if (!pdn) {
+		return;
+	}
+
+	if (pdn->callback) {
+		pdn->callback((intptr_t)cid, PDN_EVENT_CNEC_EMM, emm_err);
+	}
+
+	if (cid == pdn_act_notif.cid) {
+		pdn_act_notif.esm = emm_err;
+		k_sem_give(&pdn_act_notif.sem_cnec);
```

Comment:
This causes an EMM cause to be returned by `pdn_activate()` in the `esm` parameter. There's no way to differentiate between EMM and ESM causes, and the values are (partially) overlapping, so something probably needs to be changed.

BTW, ESM causes in esm.c are defined as hexadecimals and EMM causes in emm.c as decimals. I'm not saying these need to be aligned, just noticed when I checked whether the causes overlap or not.